### PR TITLE
Fix issue while expanding children #771

### PR DESCRIPTION
**Pull request by maxbruecken**
Fixes issue #771

### CODE REVIEW

**Review by watfordgnf — COMMENTED** [review]
Few questions.

File: src/Simple.OData.Client.Core/ODataOrderByColumn.cs [hunk 1]
```diff
@@ -14,5 +14,28 @@ public ODataOrderByColumn(string name, bool descending)
             Name = name;
             Descending = descending;
         }
+
+        public bool Equals(ODataOrderByColumn other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
+            return Name == other.Name && Descending == other.Descending;
```
[the remark this answers is not in the document]
Comment (maxbruecken) [reply src/Simple.OData.Client.Core/ODataOrderByColumn.cs:22]: The property names are case sensitive per default.

File: src/Simple.OData.Client.Core/Expressions/ExpandExpression.cs [hunk 2]
```diff
@@ -34,11 +34,19 @@ public ExpandExpressionVisitor(ITypeCache typeCache)
...
         {
-            var memberName = _typeCache.GetMappedName(node.Expression.Type, node.Member.Name);
-
-            ExpandAssociations.AddRange(node.Expression is MemberExpression
-                ? ExtractNestedExpandAssociations(node.Expression)
-                : new[] { new ODataExpandAssociation(memberName) });
+            var currentMemberExpression = node;
+            var memberName = _typeCache.GetMappedName(currentMemberExpression.Expression.Type, currentMemberExpression.Member.Name);
+            var expandAssociation = new ODataExpandAssociation(memberName);
+            while (currentMemberExpression.Expression is MemberExpression memberExpression)
```
[the remark this answers is not in the document]
Comment (maxbruecken) [reply src/Simple.OData.Client.Core/Expressions/ExpandExpression.cs]: ExtractNestedExpandAssociation did not created chain of expand associations properly (root -> child -> ...). Only most top association (root) was added.

**Comment by object** [comment]
Thank you, looks good, merging it.

**Pull request merged**